src/cmakeTaskProvider.ts

```diff
@@ -182,29 +207,32 @@ class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.OutputConsu
         let command: proc.BuildCommand | null;
         let cmakePath: string = "CMake.EXE";
         let args: string[] = [];
+        const cmakeTools: CMakeTools | undefined = getCMakeToolsForActiveFolder();
+        const cmakeDriver: CMakeDriver | undefined = (await cmakeTools?.getCMakeDriverInstance()) || undefined;
 
-        if (this.cmakeDriver) {
-            if (await this.cmakeDriver.checkNeedsReconfigure()) {
-                const result: number | undefined =  await vscode.commands.executeCommand('cmake.configure');
-                if (result !== 0) {
-                    this.writeEmitter.fire(localize("configure.finished.with.error", "Configure finished with error(s).") + endOfLine);
-                    this.closeEmitter.fire(result ? result : -1);
-                    return;
-                }
+        if (cmakeDriver) {
+            if (!this.options) {
+                this.options = {};
             }
-            command = await this.cmakeDriver.getCMakeBuildCommand(this.definedTargets ? this.definedTargets : this.defaultTargets);
-            if (command) {
-                cmakePath = command.command;
-                args = command.args ? command.args : [];
-                this.options.environment = command.build_env;
+            if (this.preset && this.preset.length !== 0) {
+                cmakePath = cmakeDriver.getCMakeCommand();
+                args = [ "--build", "--preset", this.preset];
```

Comment:
Are we calling `expandString` on `this.preset` somewhere?  I didn't see it.

Reply:
yes, it is already resolved in resolveTask. When the CustomBuildTaskTerminal is initiated, the resolved preset name is passed to it.